lib/src/main/java/org/team1540/rooster/drive/pipeline/Input.java

```diff
@@ -0,0 +1,44 @@
+package org.team1540.rooster.drive.pipeline;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import org.team1540.rooster.util.Executable;
+
+/**
+ * Extension of a {@link Supplier} adding additional composition methods.
+ *
+ * {@code Input} can be used exactly like {@link Supplier} (and library functions should not take
+ * {@code Outputs} as method parameters, instead using {@link Supplier}). However, library functions
```

Comment:
```suggestion
 * {@code Outputs} as method parameters, instead using {@link Supplier}.) However, library functions
```

Reply:
Period outside the parentheses is correct here I think—there is not a full sentence in the parentheses